File: src/main/java/net/benwoodworth/fastcraft/core/api/gui/Gui.java
package net.benwoodworth.fastcraft.core.api.gui;

import net.benwoodworth.fastcraft.core.api.dependencies.inventory.FcInventory;
import net.benwoodworth.fastcraft.core.api.gui.layouts.GuiLayout;

/**
 * A user interface for
 * @param <TItem>
 */
public class Gui<TItem> extends GuiLayout<TItem> {

    private GuiApi<TItem> guiApi;
    private FcInventory<TItem> inventory;

    /**
     * Create a new GUI.
     *
     * @param guiApi The {@link GuiApi}.
     * @param inventory the containing inventory
     */
    public Gui(GuiApi<TItem> guiApi, FcInventory<TItem> inventory) {
        this.guiApi = guiApi;
        this.inventory = inventory;
    }

    @Override
    public int getWidth() {
        return inventory.getItemGrid().getWidth();
    }

    @Override
    public int getHeight() {
        return inventory.getItemGrid().getHeight();
    }
}

File: src/main/kotlin/net/benwoodworth/fastcraft/core/api/gui/Gui.kt
package net.benwoodworth.fastcraft.core.api.gui

import net.benwoodworth.fastcraft.core.api.gui.layouts.GuiLayout
import net.benwoodworth.fastcraft.core.dependencies.item.inventory.ChestInventory

/**
 * A user interface for in-game players.
 */
class Gui(
        /*private val*/ inventory: ChestInventory
) : GuiLayout(inventory.width, inventory.height) {

    /**
     * Update the inventory to reflect the GUI.
     */
    private fun updateInventory() {
        for (x in 0 until width) {
            for (y in 0 until height) {
                // TODO Fix
                // inventory.setItem(x, y, getButton(x, y)?.buttonItem)
            }
        }
    }
}
